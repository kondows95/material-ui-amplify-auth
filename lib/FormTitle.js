"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var Box_1 = __importDefault(require("@material-ui/core/Box"));
var LockOutlined_1 = __importDefault(require("@material-ui/icons/LockOutlined"));
var FormTitle = function (props) {
    var copyProps = {};
    for (var _i = 0, _a = Object.keys(props); _i < _a.length; _i++) {
        var key = _a[_i];
        if (key !== 'label') {
            copyProps[key] = props[key];
        }
    }
    return (react_1.default.createElement(Box_1.default, __assign({}, copyProps, { display: "flex", flexDirection: "column", alignItems: "center" }),
        react_1.default.createElement(Box_1.default, { width: 40, height: 40, borderRadius: "50%", bgcolor: "secondary.main" },
            react_1.default.createElement(Box_1.default, { m: 1, color: "secondary.contrastText" },
                react_1.default.createElement(LockOutlined_1.default, null))),
        react_1.default.createElement(Box_1.default, { fontSize: "h5.fontSize", mt: 1, mb: 2 }, props.label)));
};
exports.default = FormTitle;
