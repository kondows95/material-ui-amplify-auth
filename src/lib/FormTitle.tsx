import React from 'react';
import Box from '@material-ui/core/Box';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';

type Props = {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
    label: string;
};

type CopyProps = {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
};

const FormTitle: React.FC<Props> = (props) => {
    const copyProps: CopyProps = {};
    for (const key of Object.keys(props)) {
        if (key !== 'label') {
            copyProps[key] = props[key];
        }
    }

    return (
        <Box {...copyProps} display="flex" flexDirection="column" alignItems="center">
            <Box width={40} height={40} borderRadius="50%" bgcolor="secondary.main">
                <Box m={1} color="secondary.contrastText">
                    <LockOutlinedIcon />
                </Box>
            </Box>
            <Box fontSize="h5.fontSize" mt={1} mb={2}>
                {props.label}
            </Box>
        </Box>
    );
};

export default FormTitle;
