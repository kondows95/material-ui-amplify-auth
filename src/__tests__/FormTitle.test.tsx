import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import TestLib from 'react-testing-library-wrapper';
import FormTitle from '../lib/FormTitle';

describe('Basic Usage', () => {
    let tLib: TestLib;
    beforeEach(() => {
        tLib = new TestLib(<FormTitle label="MyTitle" className="MyClass" data-testid="myId" />);
    });

    it('You can add data-testid', () => {
        tLib.get('myId');
    });

    it('You can add className', () => {
        expect(tLib.get('myId').getAttribute('class')).toContain('MyClass');
    });

    it('You can set label', () => {
        expect(tLib.render.container).toHaveTextContent('MyTitle');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});
