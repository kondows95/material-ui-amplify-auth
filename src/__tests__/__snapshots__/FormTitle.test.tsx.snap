// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Usage Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-44 MyClass"
    data-testid="myId"
  >
    <div
      class="MuiBox-root MuiBox-root-45"
    >
      <div
        class="MuiBox-root MuiBox-root-46"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-56"
    >
      MyTitle
    </div>
  </div>
</DocumentFragment>
`;
